### ISSUE TITLE
About Poor-quality data

### ISSUE THREAD

**Issue by hcrisCopy** [song2yu/Mono2Stereo#12]
Hello, regarding the #8 previously mentioned poor-quality data, could you please provide some detailed examples?

**Comment by hcrisCopy** [comment]
I'm a bit confused because in the data_process.py file, it shows "width // 2", but why does the "whitespace are not strictly aligned" error occur?

**Comment by hcrisCopy** [comment]
I noticed that in the mono2stereo test set, there are such left and right views. The black border seems to also need to be symmetrical? Could you please tell me if you took this factor into consideration when conducting the evaluation? And does the training set also contain such data? Is this kind of data considered low-quality data?
<img width="640" height="360" alt="Image" src="https://github.com/user-attachments/assets/ec467edc-9716-48fe-9de4-2b246baca37d" />
<img width="640" height="360" alt="Image" src="https://github.com/user-attachments/assets/52e8f6ca-0481-4728-897f-b9ccb3a60203" />

**Comment by hcrisCopy** [comment]
I noticed that there is also such a left-right view in the mono2stereo test set, and it seems to be completely black? Could you please tell me if you took this factor into consideration when conducting the evaluation? And does the training set also contain such data? Is this kind of data considered low-quality data?
<img width="640" height="360" alt="Image" src="https://github.com/user-attachments/assets/6a69668d-e9cd-4c4e-ae92-1258479ceec8" />

**Comment by song2yu** [comment]
Training set: We didn't apply heavy filtering at the time. We later found that some samples contain subtitles or asymmetric black borders, so we'd suggest filtering these out on your end before training.

Test set: The test set wasn't filtered against these low-quality cases either — this is indeed a limitation of the current release. Feel free to skip/ignore those samples during evaluation; we think that's a reasonable workaround.

**Comment by hcrisCopy** [comment]
> Training set: We didn't apply heavy filtering at the time. We later found that some samples contain subtitles or asymmetric black borders, so we'd suggest filtering these out on your end before training.
> 
> Test set: The test set wasn't filtered against these low-quality cases either — this is indeed a limitation of the current release. Feel free to skip/ignore those samples during evaluation; we think that's a reasonable workaround.

Could you please specify what form the "asymmetric black border" is?

**Comment by song2yu** [comment]
"Asymmetric" here refers to the cases where, after cropping, the left and right images end up with black borders of different widths. This kind of data should be fairly uncommon.

**Comment by hcrisCopy** [comment]
> "Asymmetric" here refers to the cases where, after cropping, the left and right images end up with black borders of different widths. This kind of data should be fairly uncommon.
> I noticed that in the mono2stereo test set, there are such left and right views. The black border seems to also need to be symmetrical? Could you please tell me if you took this factor into consideration when conducting the evaluation? And does the training set also contain such data? Is this kind of data considered low-quality data? <img alt="Image" width="640" height="360" src="https://private-user-images.githubusercontent.com/270675614/584198642-ec467edc-9716-48fe-9de4-2b246baca37d.jpg?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzczMDAzMzIsIm5iZiI6MTc3NzMwMDAzMiwicGF0aCI6Ii8yNzA2NzU2MTQvNTg0MTk4NjQyLWVjNDY3ZWRjLTk3MTYtNDhmZS05ZGU0LTJiMjQ2YmFjYTM3ZC5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwNDI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDQyN1QxNDI3MTJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lZmQwNzA3ZDhhNzNkNjlkNzhjN2E1NDIyNGY0NjBhYWNkZjg4M2Q4NTBkMTA2YTE2ZTYwN2EzODgwNjQ2YzAyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZyZXNwb25zZS1jb250ZW50LXR5cGU9aW1hZ2UlMkZqcGVnIn0.0vQP_doo4YVefYnu_qQ9tSTVSKkQFe3Z2IGjhWeAlXM"> <img alt="Image" width="640" height="360" src="https://private-user-images.githubusercontent.com/270675614/584198539-52e8f6ca-0481-4728-897f-b9ccb3a60203.jpg?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzczMDAzMzIsIm5iZiI6MTc3NzMwMDAzMiwicGF0aCI6Ii8yNzA2NzU2MTQvNTg0MTk4NTM5LTUyZThmNmNhLTA0ODEtNDcyOC04OTdmLWI5Y2NiM2E2MDIwMy5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwNDI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDQyN1QxNDI3MTJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mMTZiNTUwNmVhY2QwMjY3NWMxMzljZmQ5ZmZmYjJiYjIwZTIxMmI3YzJkYjFiNDUyMzBmYTBjNTUyMDAwOTY2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZyZXNwb25zZS1jb250ZW50LXR5cGU9aW1hZ2UlMkZqcGVnIn0.viukd9_OH0KdtOy41T16dYkjD84kCt6MIEwMxG5kEc0">


What puzzles me is: Is this "black border" in the picture normal? Logically speaking, a person's visual perception should not consider shifting the black border to the left and right in a symmetrical manner.

**Comment by song2yu** [comment]
> > "Asymmetric" here refers to the cases where, after cropping, the left and right images end up with black borders of different widths. This kind of data should be fairly uncommon.
> > I noticed that in the mono2stereo test set, there are such left and right views. The black border seems to also need to be symmetrical? Could you please tell me if you took this factor into consideration when conducting the evaluation? And does the training set also contain such data? Is this kind of data considered low-quality data? <img alt="Image" width="640" height="360" src="https://private-user-images.githubusercontent.com/270675614/584198642-ec467edc-9716-48fe-9de4-2b246baca37d.jpg?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzczMDAzMzIsIm5iZiI6MTc3NzMwMDAzMiwicGF0aCI6Ii8yNzA2NzU2MTQvNTg0MTk4NjQyLWVjNDY3ZWRjLTk3MTYtNDhmZS05ZGU0LTJiMjQ2YmFjYTM3ZC5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwNDI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDQyN1QxNDI3MTJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lZmQwNzA3ZDhhNzNkNjlkNzhjN2E1NDIyNGY0NjBhYWNkZjg4M2Q4NTBkMTA2YTE2ZTYwN2EzODgwNjQ2YzAyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZyZXNwb25zZS1jb250ZW50LXR5cGU9aW1hZ2UlMkZqcGVnIn0.0vQP_doo4YVefYnu_qQ9tSTVSKkQFe3Z2IGjhWeAlXM"> <img alt="Image" width="640" height="360" src="https://private-user-images.githubusercontent.com/270675614/584198539-52e8f6ca-0481-4728-897f-b9ccb3a60203.jpg?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzczMDAzMzIsIm5iZiI6MTc3NzMwMDAzMiwicGF0aCI6Ii8yNzA2NzU2MTQvNTg0MTk4NTM5LTUyZThmNmNhLTA0ODEtNDcyOC04OTdmLWI5Y2NiM2E2MDIwMy5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwNDI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDQyN1QxNDI3MTJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mMTZiNTUwNmVhY2QwMjY3NWMxMzljZmQ5ZmZmYjJiYjIwZTIxMmI3YzJkYjFiNDUyMzBmYTBjNTUyMDAwOTY2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZyZXNwb25zZS1jb250ZW50LXR5cGU9aW1hZ2UlMkZqcGVnIn0.viukd9_OH0KdtOy41T16dYkjD84kCt6MIEwMxG5kEc0">
> 
> What puzzles me is: Is this "black border" in the picture normal? Logically speaking, a person's visual perception should not consider shifting the black border to the left and right in a symmetrical manner.

The "asymmetry" I mentioned does not come from the stereo video itself. Rather, when the left and right views are placed side-by-side, there may be a slight offset between them — meaning the exact boundary between the left and right images is not necessarily at width // 2.

However, in our preprocessing we simply split the frame at width // 2 for simplicity, which introduces a small amount of asymmetric samples. You can either ignore them and train directly (this is exactly how Mono2Stereo was trained), or filter them out beforehand. Both options are fine.
